packages/notte-browser/src/notte_browser/session.py

```
@@ -821,25 +845,32 @@ async def ascrape(self, **params: Unpack[ScrapeParamsDict]) -> StructuredData[Ba
 
-        # Return data as before (backward compatible)
+        # Return data
         if data.images is not None:
             return data.images
-        if data.structured is not None:
-            if isinstance(data.structured.data, RootModel):
-                # automatically unwrap the root model otherwise it makes it unclear for the user
-                data.structured.data = data.structured.data.root  # pyright: ignore [reportUnknownMemberType, reportAttributeAccessIssue]
+        if is_structured_scrape:
+            if data.structured is None:
+                raise ScrapeFailedError("Failed to extract structured data")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)